bin/firebase-bolt

```diff
@@ -15,24 +15,130 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+'use strict';
 
+var fs = require('fs');
+var parseArgs = require('minimist');
+
+var util = require('../lib/util');
 var bolt = require('../lib/bolt');
+var pkg = require('../package.json');
+
+var VERSION_STRING = "Firebase Bolt v" + pkg.version;
+
+var opts = {
+  boolean: ['version', 'help'],
+  alias: {
+    'v': 'version',
+    'h': 'help',
+    'o': 'output',
+  },
+  unknown: function(flag) {
+    if (flag.slice(0, 2) == '--') {
+      log("Unknown flag: '" + flag + "'");
+      usage(1);
+    }
+  }
+};
+
+function main() {
+  var args = parseArgs(process.argv.slice(2), opts);
+
+  if (args.version) {
+    console.log(VERSION_STRING);
+    return;
+  }
+
+  if (args.help) {
+    usage(0);
+  }
+
+  if (args._.length > 1) {
+    log("Can only compile a single file.");
+    usage(1);
+  }
+
+  // Pipe Bolt file from stdin
+  if (args._.length === 0) {
+    readFile(process.stdin, function(data) {
```

Comment:
You could use `fs.createReadStream` to create a stream from `inFile` if it exists, then you'd have an identical code path for the parsing since both `process.stdin` and the file stream would be streams.